### PR TITLE
dataclass-wizard: init at 0.22.2

### PR DESCRIPTION
**Pull request by codifryed**
## Description of changes

The dataclass-wizard python package & application: https://github.com/rnag/dataclass-wizard

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/dataclass-wizard/default.nix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    typing-extensions
+  ] ++ lib.optionals withOptionalDependencies (builtins.attrValues {
+    inherit (passthru.optional-dependencies) all;
+  });
+
+  passthru.optional-dependencies = {
+    all = [
+      pytimeparse
+      pyyaml
+    ];
```
Comment (natsukium) [inline pkgs/development/python-modules/dataclass-wizard/default.nix:36]:
```suggestion
    timedelta = [
      pytimeparse
    ];
    yaml = [
      pyyaml
    ];
```
https://github.com/rnag/dataclass-wizard/blob/e3fe0760422613eaf79e0236840a50f27caec064/setup.py#L75-L78

**Review by natsukium — CHANGES_REQUESTED** [review]
Thanks for your contribution!
Please check some comments.

File: maintainers/maintainer-list.nix
[the remark this answers is not in the document]
Comment (codifryed) [reply maintainers/maintainer-list.nix:1]: sure, np.

File: pkgs/development/python-modules/dataclass-wizard/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    "test_frozenset"
+    "test_set"
+    "date_times_with_custom_pattern"
+    "from_dict_handles_identical_cased_json_keys"
+  ];
+
+  pythonImportsCheck = [ "dataclass_wizard" ];
+
+  meta = with lib; {
+    description = "This library provides a set of simple, yet elegant wizarding tools for interacting with the Python dataclasses module.";
```
[the remark this answers is not in the document]
Comment (codifryed) [reply pkgs/development/python-modules/dataclass-wizard/default.nix:58]: whoops, thanks.

File: pkgs/development/python-modules/dataclass-wizard/default.nix [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+, fetchFromGitHub
+, buildPythonPackage
+, pythonOlder
+, pythonAtLeast
+, pytimeparse
+, pyyaml
+, pytestCheckHook
+, pytest-mock
+, typing-extensions
+, withOptionalDependencies ? false
```
Comment (natsukium) [inline pkgs/development/python-modules/dataclass-wizard/default.nix:11]:
```suggestion
```
Besides this, LGTM.

File: pkgs/top-level/python-packages.nix [hunk 4]
```diff
@@ -2427,6 +2427,8 @@ self: super: with self; {
 
   dataclasses-serialization = callPackage ../development/python-modules/dataclasses-serialization { };
 
+  dataclass-wizard = callPackage ../development/python-modules/dataclass-wizard { };
```
Comment (natsukium) [inline pkgs/top-level/python-packages.nix:2430]: Sorry, this should be before `dataclasses-json`.

Comment (codifryed) [reply pkgs/top-level/python-packages.nix]: ah yeah

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**